src/core/addonManager.js

```diff
@@ -123,15 +125,18 @@ export function install(
   return _mozAddonManager.createInstall({ url, hash }).then((installObj) => {
     const callback = (e) => eventCallback(installObj, e);
     for (const event of INSTALL_EVENT_LIST) {
-      log.info(`[install] Adding listener for ${event}`);
+      _log.info(`[install] Adding listener for ${event}`);
       installObj.addEventListener(event, callback);
     }
     return new Promise((resolve, reject) => {
       installObj.addEventListener('onInstallEnded', () => resolve());
       installObj.addEventListener('onInstallFailed', () => reject());
-      log.info('Events to handle the installation initialized.');
-      // See: https://github.com/mozilla/addons-frontend/issues/8633
-      installObj.install().catch(reject);
+      _log.info('Events to handle the installation initialized.');
+      installObj.install().catch((error) => {
+        // See: https://github.com/mozilla/addons-frontend/issues/8668
+        _log.warn(`Ignoring error during installation: ${error}`);
+        resolve();
```

Comment:
Does this really need to resolve? Wasn't the UI working before? As I recall, the UI was working as we want it to, only an error was getting logged to Sentry, right? I think this shouldn't resolve the promise because we already get `onInstallEnded` for that.

Reply:
How should we catch that in the test case then?

Reply:
Can't you trigger `onInstallEnded` somehow?

Reply:
You could use [`Promise.race()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race) against `onInstallEnded` to make sure the rejection does nothing.

Reply:
> Can't you trigger `onInstallEnded` somehow?

ah, maybe..

Reply:
Looks like I can't.. 

No having `resolve()` or `reject()` works but it is not tested :|

Reply:
Given how this `Promise` is constructed, I think we need to either call `resolve` or `reject` though..

Reply:
So, there is one way to make it testable but it is... interesting:

```diff
diff --git a/src/core/addonManager.js b/src/core/addonManager.js
index 0247f0089..efa35d9f0 100644
--- a/src/core/addonManager.js
+++ b/src/core/addonManager.js
@@ -132,10 +132,17 @@ export function install(
       installObj.addEventListener('onInstallEnded', () => resolve());
       installObj.addEventListener('onInstallFailed', () => reject());
       _log.info('Events to handle the installation initialized.');
+
       installObj.install().catch((error) => {
+        // The `mozAddonManager` has events we can listen to, this error occurs
+        // when a user cancels the installation but we are already notified via
+        // `onInstallCancelled`.
         // See: https://github.com/mozilla/addons-frontend/issues/8668
-        _log.warn(`Ignoring error during installation: ${error}`);
-        resolve();
+        _log.warn(`Ignoring promise rejection during installation: ${error}`);
+
+        // eslint-disable-next-line no-param-reassign
+        error.SHOULD_BE_IGNORED = true;
+        reject(error);
       });
     });
   });
diff --git a/src/core/installAddon.js b/src/core/installAddon.js
index dab38ad2c..6ed6fec99 100644
--- a/src/core/installAddon.js
+++ b/src/core/installAddon.js
@@ -493,9 +493,13 @@ export class WithInstallHelpers extends React.Component<WithInstallHelpersIntern
         }
       })
       .catch((error) => {
-        _log.error(`Install error: ${error}`);
+        if (error && !!error.SHOULD_BE_IGNORED) {
+          _log.debug(`Ignored error: ${error}`);
+        } else {
+          _log.error(`Install error: ${error}`);
 
-        dispatch(setInstallError({ guid, error: FATAL_INSTALL_ERROR }));
+          dispatch(setInstallError({ guid, error: FATAL_INSTALL_ERROR }));
+        }
       });
   }

```

I think I'd be fine with not calling `reject`/`resolve` and therefore remove the test case for it. That would avoid the complexity above ^

This is needed because `resolve`/`reject` are (1) required for the test case, (2) used in `installAddon`.

Reply:
This is probably terrible but you could do it like this 😆 

```diff
diff --git a/src/core/addonManager.js b/src/core/addonManager.js
index 0247f0089..cf68c68e4 100644
--- a/src/core/addonManager.js
+++ b/src/core/addonManager.js
@@ -135,7 +135,6 @@ export function install(
       installObj.install().catch((error) => {
         // See: https://github.com/mozilla/addons-frontend/issues/8668
         _log.warn(`Ignoring error during installation: ${error}`);
-        resolve();
       });
     });
   });
diff --git a/tests/unit/core/test_addonManager.js b/tests/unit/core/test_addonManager.js
index f7436c4c4..3ed91063c 100644
--- a/tests/unit/core/test_addonManager.js
+++ b/tests/unit/core/test_addonManager.js
@@ -199,17 +199,28 @@ describe(__filename, () => {
 
     // See: https://github.com/mozilla/addons-frontend/issues/8633
     it('logs and ignores rejected install errors', async () => {
-      const _log = getFakeLogger();
+      let finishLogWarn;
+      const logWarnWasCalled = new Promise((resolve) => {
+        finishLogWarn = resolve;
+      });
+
+      const _log = getFakeLogger({
+        warn: sinon.spy(() => finishLogWarn()),
+      });
+
       fakeInstallObj.install = sinon.spy(() => {
         return Promise.reject(new Error('oops'));
       });
 
-      await addonManager.install(fakeInstallUrl, fakeCallback, {
-        _log,
-        _mozAddonManager: fakeMozAddonManager,
-        src: 'home',
-      });
+      addonManager
+        .install(fakeInstallUrl, fakeCallback, {
+          _log,
+          _mozAddonManager: fakeMozAddonManager,
+          src: 'home',
+        })
+        .then(unexpectedSuccess);
 
+      await logWarnWasCalled;
       sinon.assert.calledOnce(_log.warn);
     });
 
```

If you think putting a `resolve()` in there is harmless then it would ok to ship it. I'm just wondering what the side effect is of calling `resolve()` on every promise rejection would be since we haven't done that in the past.

Reply:
No, you were right with the `promise`/`resolve`. It has side effects, i.e. `resolve` will trigger a GA event (but not always it seems) and `reject` will... display the banner we try to hide here.

Reply:
I think I like your solution better than mine

Reply:
Maybe this is a little cleaner since it's not dependent on `log.warn()`:

```diff
diff --git a/src/core/addonManager.js b/src/core/addonManager.js
index 0247f0089..5998fcbf6 100644
--- a/src/core/addonManager.js
+++ b/src/core/addonManager.js
@@ -105,6 +105,7 @@ type OptionalInstallParams = {|
   _log?: typeof log,
   src: string,
   hash?: string | null,
+  onIgnoredRejection?: () => void,
 |};
 
 export function install(
@@ -114,6 +115,7 @@ export function install(
     _log = log,
     _mozAddonManager = window.navigator.mozAddonManager,
     hash,
+    onIgnoredRejection = () => {},
     src,
   }: OptionalInstallParams = {},
 ) {
@@ -135,7 +137,7 @@ export function install(
       installObj.install().catch((error) => {
         // See: https://github.com/mozilla/addons-frontend/issues/8668
         _log.warn(`Ignoring error during installation: ${error}`);
-        resolve();
+        onIgnoredRejection();
       });
     });
   });
diff --git a/tests/unit/core/test_addonManager.js b/tests/unit/core/test_addonManager.js
index f7436c4c4..89fa2a566 100644
--- a/tests/unit/core/test_addonManager.js
+++ b/tests/unit/core/test_addonManager.js
@@ -199,17 +199,27 @@ describe(__filename, () => {
 
     // See: https://github.com/mozilla/addons-frontend/issues/8633
     it('logs and ignores rejected install errors', async () => {
+      let finishInstall;
+      const installToFinish = new Promise((resolve) => {
+        finishInstall = resolve;
+      });
+
       const _log = getFakeLogger();
+
       fakeInstallObj.install = sinon.spy(() => {
         return Promise.reject(new Error('oops'));
       });
 
-      await addonManager.install(fakeInstallUrl, fakeCallback, {
-        _log,
-        _mozAddonManager: fakeMozAddonManager,
-        src: 'home',
-      });
+      addonManager
+        .install(fakeInstallUrl, fakeCallback, {
+          _log,
+          _mozAddonManager: fakeMozAddonManager,
+          onIgnoredRejection: () => finishInstall(),
+          src: 'home',
+        })
+        .then(unexpectedSuccess);
 
+      await installToFinish;
       sinon.assert.calledOnce(_log.warn);
     });
 
```

It's a little fragile, though, because `addonManager.install()` could cause an uncaught rejection in this test (after a refactoring mistake).

Reply:
> It's a little fragile, though, because addonManager.install() could cause an uncaught rejection in this test (after a refactoring mistake).

Depending on `log.warn` is what we want, no? So maybe your initial proposal is good enough?

Reply:
I dunno, it felt hacky to overload the implementation of `log.warn()` because that may not be clear when reading the `install()` function without having knowledge of the test. By using an explicit `onIgnoredRejection()` it is more clear that something can happen in that rejection. It gives the reader something to grep for, at least.